diff --git a/ubuntu1804_install_dependencies.sh b/ubuntu1804_install_dependencies.sh
index 44ed98a..56dec82 100755
--- a/ubuntu1804_install_dependencies.sh
+++ b/ubuntu1804_install_dependencies.sh
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
-sudo apt-get install \
+sudo DEBIAN_FRONTEND=noninteractive apt-get install -y \
     git \
     cmake \
     build-essential \
@@ -18,11 +18,12 @@ sudo apt-get install \
     libglew-dev \
     qtbase5-dev \
     libqt5opengl5-dev \
-    libcgal-dev
+    libcgal-dev \
+    libcgal-qt5-dev
 
-sudo apt-get install nvidia-driver-390 nvidia-utils-390 nvidia-cuda-toolkit
+sudo DEBIAN_FRONTEND=noninteractive apt-get install nvidia-driver-410 nvidia-utils-410 nvidia-cuda-toolkit -y
 
-sudo apt-get install libatlas-base-dev libsuitesparse-dev
+sudo DEBIAN_FRONTEND=noninteractive apt-get install libatlas-base-dev libsuitesparse-dev -y
 git clone https://ceres-solver.googlesource.com/ceres-solver
 cd ceres-solver
 # git checkout $(git describe --tags) # Checkout the latest release
