diff --git a/Reconstruction/code/CMakeLists.txt b/Reconstruction/code/CMakeLists.txt
index 417a4fc..88e576b 100644
--- a/Reconstruction/code/CMakeLists.txt
+++ b/Reconstruction/code/CMakeLists.txt
@@ -28,6 +28,10 @@ set(header_files
 
 set(CMAKE_CXX_STANDARD:STRING 11)
 
+find_package(VTK REQUIRED)
+include_directories(${VTK_INCLUDE_DIRS})
+link_directories(${VTK_LIBRARY_DIRS})
+
 find_package(PCL)
 include_directories(${PCL_INCLUDE_DIRS})
 include_directories(${PCL_DIR}/include)
@@ -37,5 +41,5 @@ find_library(CGAL_LIBRARY CGAL)
 find_library(GMP_LIBRARY gmp)
 
 add_executable(${recon_app_name} Reconstruction_main.cpp ${src_files} ${header_files})
-target_link_libraries(${recon_app_name} ${PCL_LIBRARIES} ${CGAL_LIBRARY} ${GMP_LIBRARY})
+target_link_libraries(${recon_app_name} ${PCL_LIBRARIES} ${CGAL_LIBRARY} ${GMP_LIBRARY} ${VTK_LIBRARIES})
 install(TARGETS ${recon_app_name} DESTINATION bin)
diff --git a/Segmentation/code/CMakeLists.txt b/Segmentation/code/CMakeLists.txt
index 5cade6b..59d2cd0 100644
--- a/Segmentation/code/CMakeLists.txt
+++ b/Segmentation/code/CMakeLists.txt
@@ -43,7 +43,13 @@ find_package(GMP REQUIRED)
 
 link_directories("${PCL_LIBRARY_DIRS}")
 
+# This comes from liblas, as it is not declared public there
+find_package(Boost COMPONENTS program_options REQUIRED)
+include_directories(${Boost_INCLUDE_DIR})
+link_directories(${Boost_LIBRARY_DIR})
+
+
 add_executable(${seg_app_name} Segmentation_main.cpp ${src_files} ${header_files})
-target_link_libraries(${seg_app_name} ${PCL_LIBRARIES} ${GDAL_LIBRARY} ${libLAS_LIBRARIES} ${CGAL_LIBRARY} ${GMP_LIBRARIES})
+target_link_libraries(${seg_app_name} ${PCL_LIBRARIES} ${GDAL_LIBRARY} ${libLAS_LIBRARIES} ${CGAL_LIBRARY} ${GMP_LIBRARIES} ${Boost_LIBRARIES})
 
 install(TARGETS ${seg_app_name} DESTINATION bin)
