diff --git a/apps/CMakeLists.txt b/apps/CMakeLists.txt
index beb83e2..b01210c 100644
--- a/apps/CMakeLists.txt
+++ b/apps/CMakeLists.txt
@@ -107,7 +107,14 @@ endif()
 if(LAS2COL)
     set(LAS2COL_SRC lascommon.c ${LAS2COL}.c)
     add_executable(${LAS2COL} ${LAS2COL_SRC})
-    target_link_libraries(${LAS2COL} ${LIBLAS_C_LIB_NAME})
+    if (${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION} GREATER 3.1)
+        set(THREADS_PREFER_PTHREAD_FLAG ON)
+        find_package(Threads REQUIRED)
+        target_link_libraries(${LAS2COL} ${LIBLAS_C_LIB_NAME} Threads::Threads)
+    else (${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION} GREATER 3.1)
+        find_package(Threads REQUIRED)
+        target_link_libraries(${LAS2COL} ${LIBLAS_C_LIB_NAME} "${CMAKE_THREAD_LIBS_INIT}")
+    endif (${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION} GREATER 3.1)
 endif()
 
 # Build las2pg
